protocol/deploys/99-example-deploy-script.ts

```diff
@@ -0,0 +1,42 @@
+import '@nomiclabs/hardhat-ethers';
+import { ethers, network } from 'hardhat';
+
+import { DEPLOYED_CONTRACTS } from '../contract-addresses';
+
+import { deployAndMine, ensureExpectedEnvvars, mine } from './helpers';
+import { BigNumber } from 'ethers';
+
+async function main() {
+  const deployedContracts = DEPLOYED_CONTRACTS[network.name];
+  const args: {
+    CONSTRUCTOR_ARG_1: string;
+    CONSTRUCTOR_ARG_2: string;
+    VESTING_DURATION: string;
+  } = {
+    CONSTRUCTOR_ARG_1: '';
+    CONSTRUCTOR_ARG_2: '';
+    VESTING_DURATION: '',
+  }
+  ensureExpectedEnvvars(args);
+
+  const [owner] = await ethers.getSigners();
+
+  const someSmartContractFactory = new SomeSmartContract__factory(owner)
+  const someSmartContract: Presale = await deployAndMine("PRESALE", someSmartContractFactory, someSmartContractFactory.deploy,
```

Comment:
To make it super crisp, I wonder if this can just take one single arg, which is an object of the required `{constructor_arg_name:value}`? Not sure if this is possible with ts magic

```
const someSmartContract: Presale = await deployAndMine(
    "PRESALE",
    someSmartContractFactory,
    someSmartContractFactory.deploy,
    args,
);
```

So the args need to match the constructor kv args (name and value type)